src/main/scala/nl/hugo/redbook/ch7/Par.scala

```diff
@@ -0,0 +1,114 @@
+package nl.hugo.redbook.ch7
+
+import java.util.concurrent._
+import language.implicitConversions
+
+object Par {
+  type Par[A] = ExecutorService => Future[A]
+
+  def run[A](s: ExecutorService)(a: Par[A]): Future[A] = a(s)
```

Comment:
IMHO `run` should return `A` instead of `Future[A]` (requiring the implementation to be `a(s).get`). There's an inconsistency in the book: on page 116 `run()` returns 'A', while on page 120 the implementation of `choice` calls `get` on it. I think the `get` should be dropped here, because otherwise the `run` method doesn't add any value (and does not make much sense IMHO).